pkg/api/handlers.go

```diff
@@ -39,6 +43,37 @@ func newHandler(context *clusterd.Context, config *Config) *Handler {
 	}
 }
 
+// RegisterMetrics registers all collected metrics by this API server.  Note this should be called in a
+// goroutine because it will retry upon failure and block until successful.
+func (h *Handler) RegisterMetrics() error {
```

Comment:
Do we always want to register with prometheus? If prometheus isn't running in the environment we will have a ceph connection sitting around.